tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_004_pos.ksh

```diff
@@ -110,11 +110,11 @@ for arg in "$fs1 $mntp1" "$clone $mntp2"; do
 	cd $mntp
 	log_mustnot $ZFS destroy $fs
 
+	cd $olddir
 	log_must $ZFS destroy -f $fs
 	datasetexists $fs && \
 		log_fail "'zfs destroy -f' fails to destroy busy filesystem."
 
-	cd $olddir
```

Comment:
The key issue here is that illumos kernel treats this case differently than the Linux kernel.  On illumos you can unmount and destroy a filesystem which is busy and all consumers of it get EIO.  On Linux the expected behavior is to prevent the unmount and destroy.

Changing that fundamental difference isn't easy, or necessarily even desirable.  Instead, how about updating the test case to reflect this reality.  Something like this:

```diff
@@ -110,10 +110,15 @@ for arg in "$fs1 $mntp1" "$clone $mntp2"; do
        cd $mntp
        log_mustnot $ZFS destroy $fs
 
-       log_must $ZFS destroy -f $fs
-       datasetexists $fs && \
-               log_fail "'zfs destroy -f' fails to destroy busy filesystem."
-
+       if is_linux; then
+               log_mustnot $ZFS destroy -f $fs
+               datasetexists $fs && \
+                       log_fail "'zfs destroy -f' destroyed busy filesystem."
+       else
+               log_must $ZFS destroy -f $fs
+               datasetexists $fs && \
+                       log_fail "'zfs destroy -f' fails to destroy busy filesystem."
+       fi
        cd $olddir
 done
 
```